garbage.go

```diff
@@ -27,64 +33,75 @@ func (m *Sealing) pledgeReader(size uint64, parts uint64) io.Reader {
 	return io.MultiReader(readers...)
 }
 
-func (m *Sealing) pledgeSector(ctx context.Context, sectorID uint64, existingPieceSizes []uint64, sizes ...uint64) ([]Piece, error) {
+func (m *Sealing) pledgeSector(ctx context.Context, sectorNum abi.SectorNumber, existingPieceSizes []abi.UnpaddedPieceSize, sizes ...abi.UnpaddedPieceSize) ([]Piece, error) {
 	if len(sizes) == 0 {
 		return nil, nil
 	}
 
-	log.Infof("Pledge %d, contains %+v", sectorID, existingPieceSizes)
+	log.Infof("Pledge %d, contains %+v", sectorNum, existingPieceSizes)
+
+	arg := &market.PublishStorageDealsParams{
+		Deals: make([]market.ClientDealProposal, len(sizes)),
+	}
 
-	pieces := make([]PieceInfo, len(sizes))
 	for i, size := range sizes {
-		commP, err := m.FastPledgeCommitment(size, uint64(runtime.NumCPU()))
+		commP, err := m.FastPledgeCommitment(size, uint64(1))
 		if err != nil {
-			return nil, err
+			return nil, handle("failed to generate pledge commitment: ", err)
 		}
 
-		pieces[i] = PieceInfo{
-			Size:  size,
-			CommP: commP,
+		sdp := market.ClientDealProposal{
+			Proposal: market.DealProposal{
+				Client:               m.worker,
```

Comment:
consider a function to retrieve worker address from state.

Reply:
considered it, and, you were right. updated.